python/dgl/data/minigc.py

```diff
@@ -76,8 +78,9 @@ def __getitem__(self, idx):
 
     def save(self):
         """save the graph list and the labels"""
-        graph_path = os.path.join(self.save_path, 'dgl_graph.bin')
-        save_graphs(str(graph_path), self.graphs, {'labels': self.labels})
+        if self.save_graph :
+            graph_path = os.path.join(self.save_path, 'dgl_graph.bin')
+            save_graphs(str(graph_path), self.graphs, {'labels': self.labels})
```

Comment:
As has_cache() is not implemented, it always return False, so each time we call use this class is actually creating new dataset and save() and load() is not used, right?

Reply:
As this is a synthetic dataset, we need to check whether the saved dataset match the given parameters. If not new dataset need to be generated.